common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java

```diff
@@ -283,29 +283,49 @@ static public boolean checkPassword(@NonNull String encodedPassword, @NonNull St
 		}
 	}
 
-	static public byte[] encryptStringToBytes(String input, String password) throws Exception {
-		SecretKeySpec skeyspec=new SecretKeySpec(password.getBytes(), ENCRYPTION_CIPHER);
-		Cipher cipher=Cipher.getInstance(ENCRYPTION_CIPHER);
+	static private byte[] encryptStringToBytes(String input, String password, Algorythm algorythm) throws Exception {
+		SecretKeySpec skeyspec=new SecretKeySpec(password.getBytes(), algorythm.name());
+		Cipher cipher=Cipher.getInstance(algorythm.name());
 		cipher.init(Cipher.ENCRYPT_MODE, skeyspec);
 		return cipher.doFinal(input.getBytes(StandardCharsets.UTF_8));
 	}
 
-	static public String decryptStringFromBytes(byte[] data, String password) throws Exception {
-		SecretKeySpec skeyspec=new SecretKeySpec(password.getBytes(), ENCRYPTION_CIPHER);
-		Cipher cipher=Cipher.getInstance(ENCRYPTION_CIPHER);
+	static private String decryptStringFromBytes(byte[] data, String password, Algorythm algorythm) throws Exception {
+		SecretKeySpec skeyspec=new SecretKeySpec(password.getBytes(), algorythm.name());
+		Cipher cipher=Cipher.getInstance(algorythm.name());
 		cipher.init(Cipher.DECRYPT_MODE, skeyspec);
 		byte[] decrypted = cipher.doFinal(data);
 		return new String(decrypted, StandardCharsets.UTF_8);
 	}
 
 	static public String encryptString(String input, String password) throws Exception {
-		byte[] bytes = encryptStringToBytes(input, password);
+		byte[] bytes = encryptStringToBytes(input, password, Algorythm.Blowfish);
 		return StringTool.encodeBase64ToString(bytes);
 	}
 
 	static public String decryptString(String input, String password) throws Exception {
 		byte[] bytes = StringTool.decodeBase64(input);
-		return decryptStringFromBytes(bytes, password);
+		return decryptStringFromBytes(bytes, password, Algorythm.Blowfish);
+	}
+
+	static public String encryptStringAes(String input, String password) throws Exception {
```

Comment:
Why not just have base64 and hex variants and pass in the algorithm? We now have to add a whole set of methods for every algorithm added...

Reply:
we had encryptString and decryptString in old code base that used Blowfish and base64. So, I had to keep it for backward compatibility. That is why I just added 2 more pairs of 2 lines methods, that I used in my code, with reusing the stuff that was there already. Isn't it better to have dedicated methods then combining the parameters in each call? I mean, can be easier to make the mistake.

Reply:
ok, I made it as asked.